src/main/QueryServer.cpp

```diff
@@ -0,0 +1,187 @@
+// Copyright 2024 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "main/QueryServer.h"
+#include "bucket/BucketListSnapshot.h"
+#include "bucket/BucketSnapshotManager.h"
+#include "ledger/LedgerTxnImpl.h"
+#include "util/Logging.h"
+#include "util/XDRStream.h" // IWYU pragma: keep
+#include <exception>
+#include <json/json.h>
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace
+{
+template <typename T>
+std::optional<T>
+parseOptionalParam(std::map<std::string, std::vector<std::string>> const& map,
+                   std::string const& key)
+{
+    auto i = map.find(key);
+    if (i != map.end())
+    {
+        if (i->second.size() != 1)
+        {
+            std::string errorMsg = fmt::format(
+                FMT_STRING("Expected exactly one '{}' argument"), key);
+            throw std::runtime_error(errorMsg);
+        }
+
+        std::stringstream str(i->second.at(0));
+        T val;
+        str >> val;
+
+        // Throw an error if not all bytes were loaded into `val`
+        if (str.fail() || !str.eof())
+        {
+            std::string errorMsg =
+                fmt::format(FMT_STRING("Failed to parse '{}' argument"), key);
+            throw std::runtime_error(errorMsg);
+        }
+        return std::make_optional<T>(val);
+    }
+
+    return std::nullopt;
+}
+}
+
+namespace stellar
+{
+QueryServer::QueryServer(const std::string& address, unsigned short port,
+                         int maxClient, size_t threadPoolSize,
+                         BucketSnapshotManager& bucketSnapshotManager)
+    : mServer(address, port, maxClient, threadPoolSize)
+{
+    LOG_INFO(DEFAULT_LOG, "Listening on {}:{} for Query requests", address,
+             port);
+
+    mServer.add404(std::bind(&QueryServer::notFound, this, _1, _2, _3));
+    addRoute("getledgerentryraw", &QueryServer::getLedgerEntryRaw);
```

Comment:
Correct, sorry for not being super clear. `getledgerentry` has been removed (since it shouldn't really have been called in the first place) and replaced with the new `getledgerentryraw`. `getledgerentryraw` only routes on the new server endpoint. All the other RPC endpoints we've discussed will also be on this new endpoint.